source/blogs/uart/uart.rst

```diff
@@ -0,0 +1,118 @@
+Celadon Serial debug logs output enable
+=======================================
+
+Table of Contents
+#################
+
+.. contents::
+    :depth: 2
+    :local:
+
+Serial debug logs output enable on KBL_NUC
+------------------------------------------
+
+HW connection
+~~~~~~~~~~~~~
+
+On KBL_NUC, HW board has a UART output, extend it out with the connector below:
+
+.. figure:: images/kbl_uart_connector.jpg
+    :align: center
+
+Connect to KBL_NUC:
+
+.. figure:: images/kbl_extend_uart.jpg
+    :align: center
+
+Then connect it to PC host with a serial2USB converter cable.
+
+.. figure:: images/usb2uart_female.jpg
+    :align: center
+
+SW support
+~~~~~~~~~~
+
+enable kernel log serial in mixins.spec for the project before image building.
+
+    .. code-block:: text
+
+	serialport: ttyS0
+
+On PC host, open the ttyUSB device to capture logs
+
+    .. code-block:: bash
+
+	sudo minicom -D /dev/ttyUSB0
+
+Status
+~~~~~~
+
+The logs from BIOS,Kernelflinger, Kernel and AOSP can be output during startup.
+
+Serial debug logs output enable on APL_NUC
+------------------------------------------
+
+HW connection
+~~~~~~~~~~~~~
+
+On APL_NUC, HW board has no UART output.For kernel and AOSP logs output, we need a cable below to translate USB to serial, ttyUSB driver is supported in kernel:
+
+.. figure:: images/usb2uart_male.jpg
+    :align: center
+
+Then connect it to PC host with a serial2USB converter cable.
+
+.. figure:: images/usb2uart_female.jpg
+    :align: center
+
+SW support
+~~~~~~~~~~
+
+Enable kernel log serial in mixins.spec for the project before image building.
+
+    .. code-block:: text
+
+	serialport: ttyUSB0
+
+On PC host, open the ttyUSB device to capture logs
+
+    .. code-block:: bash
+
+	sudo minicom -D /dev/ttyUSB0
+Because Kernelflinger has no ttyUSB driver, the logs is saved to a file.
+
+Status
+~~~~~~
+
+The logs from Kernelflinger is saved to a file. And Kernel and AOSP logs can be captured in a console, e.g.: minicom.
+
+Serial debug logs output enable on whisklake (WHL)
+--------------------------------------------------
+
+HW connection
+~~~~~~~~~~~~~
+
+On WHL,  HW board has a UART with 9-pins connector output. We just need connect it to PC host with a serial2USB converter cable.
+
+.. figure:: images/usb2uart_female.jpg
+    :align: center
+
+SW support
+~~~~~~~~~~
+
+enable kernel log serial in mixins.spec for the project before image building.
+
+    .. code-block:: text
+
+	serialport: ttyS0
+
+On PC host, open the ttyUSB device to capture logs
+
+    .. code-block:: bash
+
+	sudo minicom -D /dev/ttyUSB0
+
+Status
+~~~~~~
+
+The logs from BIOS,Kernelflinger, Kernel and AOSP can be output during startup.
```

Comment:
Suggest to rewrite "You will get the output logs from the BIOS, Kernelflonger, Linux kernel, and Celadon Android stack during the board starting up."

Reply:
Done! Thanks!